### PR TITLE
Raw Subscribe; new example_10

### PR DESCRIPTION
**Pull request by serges147**
- Implemented RAW `presentation::Subscriber<void>`.
- Added new "1_presentation/example_10_posix_udp" - demonstrates how to publish and subscribe to **RAW** (aka `void`) messages using presentation layer `Publisher<void>` and `Subscriber<void>` classes. This "example_10" made almost self-sufficient - only UDP media creation is done via helpers.
- Moved examples under "0_transport" and "1_presentation" sub-folders according to which layer they belongs to (mainly).

Also:
- sync CAVL

### CODE REVIEW

File: docs/examples/0_transport/example_02_posix_udp.cpp [hunk 1]
```diff
@@ -7,11 +7,11 @@
...
-#include "platform/common_helpers.hpp"
-#include "platform/node_helpers.hpp"
-#include "platform/posix/posix_single_threaded_executor.hpp"
-#include "platform/posix/udp/udp_media.hpp"
-#include "platform/tracking_memory_resource.hpp"
+#include "../platform/common_helpers.hpp"
+#include "../platform/node_helpers.hpp"
+#include "../platform/posix/posix_single_threaded_executor.hpp"
+#include "../platform/posix/udp/udp_media.hpp"
+#include "../platform/tracking_memory_resource.hpp"
```
Comment (pavel-kirienko) [inline docs/examples/0_transport/example_02_posix_udp.cpp:14]: ew

File: docs/examples/0_transport/example_03_linux_socketcan.cpp [hunk 2]
```diff
@@ -7,11 +7,11 @@
...
-#include "platform/common_helpers.hpp"
-#include "platform/linux/can/can_media.hpp"
-#include "platform/linux/epoll_single_threaded_executor.hpp"
-#include "platform/node_helpers.hpp"
-#include "platform/tracking_memory_resource.hpp"
+#include "../platform/common_helpers.hpp"
+#include "../platform/linux/can/can_media.hpp"
+#include "../platform/linux/epoll_single_threaded_executor.hpp"
+#include "../platform/node_helpers.hpp"
+#include "../platform/tracking_memory_resource.hpp"
```
Comment (pavel-kirienko) [inline docs/examples/0_transport/example_03_linux_socketcan.cpp:14]: Shouldn't `docs/examples/platform` be in the include search path set?

File: include/libcyphal/presentation/presentation.hpp [hunk 3]
```diff
@@ -76,34 +78,57 @@ class Presentation final : private detail::IPresentationDelegate
     ///
     /// The subscriber must never outlive this presentation object.
     ///
-    /// @tparam Message DSDL compiled (aka Nunavut generated) type of the message to subscribe.
+    /// Internally, multiple subscribers to the same subject id are using the same instance of shared RX session.
+    /// Such sharing is transparent to the user of the library, but has implications on the extent bytes parameter -
+    /// the very first subscriber to a subject id defines the extent bytes for all subscribers to that subject id.
+    /// So implicit extent bytes might be ignored if the subscriber is the second or later one to the subject id.
```
Comment (pavel-kirienko) [inline include/libcyphal/presentation/presentation.hpp:84]:
```suggestion
    /// So implicit extent bytes might be ignored if the subscriber is the second or later one to the subject id.
    /// This behavior may be improved in a future version such that the largest extent of all existing subscribers is used.
```

File: include/libcyphal/presentation/presentation.hpp [hunk 4]
```diff
@@ -76,34 +78,57 @@ class Presentation final : private detail::IPresentationDelegate
...
     }
 
+    /// @brief Makes a raw message subscriber.
+    ///
+    /// The subscriber must never outlive this presentation object.
+    ///
+    /// Internally, multiple subscribers to the same subject id are using the same instance of shared RX session.
+    /// Such sharing is transparent to the user of the library, but has implications on the extent bytes parameter -
+    /// the very first subscriber to a subject id defines the extent bytes for all subscribers to that subject id.
+    /// So explicit `extent_bytes` might be ignored if the subscriber is the second or later one to the subject id.
```
Comment (pavel-kirienko) [inline include/libcyphal/presentation/presentation.hpp:112]: Should we add the same comment here?

File: include/libcyphal/presentation/subscriber_impl.hpp [hunk 5]
```diff
@@ -139,6 +139,35 @@ class SubscriberImpl final : public cavl::Node<SubscriberImpl>, public detail::S
...
+                    // This is safe downcast b/c we know that the `curr_node` is of type `Subscriber`.
+                    // Otherwise, the `deserializer_` would be different from `this_deserializer`.
+                    auto* const subscriber = static_cast<Subscriber*>(curr_node);
+                    subscriber->onReceiveCallback(context.approx_now, context.buffer, context.metadata);
+
+                    // NB! `curr_node` or `subscriber` must not be used anymore b/c they may be invalidated
+                    // by a callback activity, f.e. by moving or freeing the `Subscriber` object.
+                }
+            }
+
```
Comment (pavel-kirienko) [inline include/libcyphal/presentation/subscriber_impl.hpp:170]: I will get back to this tomorrow

**Review by pavel-kirienko — APPROVED** [review]

**Review by lydia-at-amazon — APPROVED** [review]
Looks good, thank you!

**Pull request merged**